applications/ContactStructuralMechanicsApplication/tests/test_mortar_mapper.py

```diff
@@ -0,0 +1,224 @@
+            node.Set(KratosMultiphysics.MASTER, False)
+        del(node)
+        self.model_part_master = self.main_model_part.GetSubModelPart("Parts_Parts_Auto2")
+        for node in self.model_part_master.Nodes:
+            node.Set(KratosMultiphysics.MASTER, True)
+            node.Set(KratosMultiphysics.SLAVE, False)
+        del(node)
+        
+        for prop in self.main_model_part.GetProperties():
+            prop[ContactStructuralMechanicsApplication.INTEGRATION_ORDER_CONTACT] = 3 
+            prop[ContactStructuralMechanicsApplication.ACTIVE_CHECK_FACTOR] = 3.0e-1
+        
+        for node in self.mapping_model_part.Nodes:
+            node.Set(KratosMultiphysics.INTERFACE, True)
+            
+        Preprocess = ContactStructuralMechanicsApplication.InterfacePreprocessCondition(self.main_model_part)
+
+        interface_parameters = KratosMultiphysics.Parameters("""{"condition_name": "", "final_string": "", "simplify_geometry": false}""")
+        interface_parameters["condition_name"].SetString("ALMFrictionlessMortarContact")
+        Preprocess.GenerateInterfacePart3D(self.main_model_part, self.mapping_model_part, interface_parameters)
+            
+        # We copy the conditions to the ContactSubModelPart
+        for cond in self.mapping_model_part.Conditions:
+            interface_model_part.AddCondition(cond)    
+        del(cond)
+        for node in self.mapping_model_part.Nodes:
+            interface_model_part.AddNode(node, 0)    
+        del(node)
+
+        # We initialize the conditions    
+        alm_init_var = ContactStructuralMechanicsApplication.ALMFastInit(self.mapping_model_part) 
+        alm_init_var.Execute()
+
+        search_parameters = KratosMultiphysics.Parameters("""
+        {
+            "search_factor"               : 3.5,
+            "allocation_size"             : 1000,
+            "type_search"                 : "InRadius",
+            "use_exact_integration"       : true
+        }
+        """)
+        contact_search = ContactStructuralMechanicsApplication.TreeContactSearch(self.main_model_part, search_parameters)
+        
+        # We initialize the search utility
+        contact_search.CreatePointListMortar()
+        contact_search.InitializeMortarConditions()
+        contact_search.UpdateMortarConditions()
+
+        for node in self.model_part_slave.Nodes:
+            x = node.X 
+            y = node.Y 
+            z = node.Z
+            node.SetSolutionStepValue(KratosMultiphysics.PRESSURE, z)
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_X, x)
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_Y, y)
+            node.SetSolutionStepValue(KratosMultiphysics.VELOCITY_Z, z)
+        del(node)
+
+        if (num_nodes == 3): 
+            self.mortar_mapping_double = KratosMultiphysics.SimpleMortarMapperProcess3D3NDoubleHistorical(self.main_model_part, KratosMultiphysics.PRESSURE)
```

Comment:
question: why don't u use an interface similar to that of the AdvancedNMMapper or to the one developed by @philbucher ?

Reply:
In any case this process was just an example, tell me what I am supposed to follow